FWCore/Framework/src/ScheduleBuilder.cc

```diff
@@ -0,0 +1,728 @@
+                                                ConditionalTaskHelper const& iConditionalTaskHelper,
+                                                std::unordered_set<std::string>& oAllConditionalModules) {
+      auto modnames = ioProcessPSet.getParameter<std::vector<std::string>>(iPathName);
+      std::vector<ModuleInPath> tmpworkers;
+
+      //Pull out ConditionalTask modules
+      auto condRange = findConditionalTaskModulesRange(modnames);
+
+      std::unordered_set<std::string> conditionalmods;
+      //An EDAlias may be redirecting to a module on a ConditionalTask
+      std::unordered_multimap<std::string, edm::ProductDescription const*> conditionalModsBranches;
+      std::unordered_map<std::string, unsigned int> conditionalModOrder;
+      if (condRange.first != condRange.second) {
+        for (auto it = condRange.first; it != condRange.second; ++it) {
+          // ordering needs to skip the # token in the path list
+          conditionalModOrder.emplace(*it, it - modnames.begin() - 1);
+        }
+        //the last entry should be ignored since it is required to be "@"
+        conditionalmods = std::unordered_set<std::string>(std::make_move_iterator(condRange.first),
+                                                          std::make_move_iterator(condRange.second));
+
+        conditionalModsBranches = iConditionalTaskHelper.conditionalModuleBranches(conditionalmods);
+        modnames.erase(std::prev(condRange.first), modnames.end());
+
+        // Make a union of all conditional modules from all Paths
+        oAllConditionalModules.insert(conditionalmods.begin(), conditionalmods.end());
+      }
+
+      unsigned int placeInPath = 0;
+      for (auto const& name : modnames) {
+        //Modules except EDFilters are set to run concurrently by default
+        bool doNotRunConcurrently = false;
+        WorkerInPath::FilterAction filterAction = WorkerInPath::Normal;
+        if (name[0] == '!') {
+          filterAction = WorkerInPath::Veto;
+        } else if (name[0] == '-' or name[0] == '+') {
+          filterAction = WorkerInPath::Ignore;
+        }
+        if (name[0] == '|' or name[0] == '+') {
+          //cms.wait was specified so do not run concurrently
+          doNotRunConcurrently = true;
+        }
+
+        std::string moduleLabel = name;
+        if (filterAction != WorkerInPath::Normal or name[0] == '|') {
+          moduleLabel.erase(0, 1);
+        }
+
+        auto module = getModule(ioProcessPSet,
+                                moduleLabel,
+                                ioModuleRegistry,
+                                ioProductRegistry,
+                                iActivityRegistry,
+                                iPrealloc,
+                                iProcessConfiguration);
+        if (module == nullptr) {
+          std::string pathType("endpath");
+          if (std::count(iEndPathNames.begin(), iEndPathNames.end(), iPathName) == 0) {
```

Comment:
`std::find()` would seem more idiomatic, e.g.
```suggestion
          if (std::ranges::find(iEndPathNames, iPathName) == iEndPathNames.end()) {
```